### PR TITLE
Issue #15236: format some ITs Google Inputs by formatter

### PR DESCRIPTION
**Pull request by romani**
Issue #15236

Exact list of Inputs to validate is separate conversation.
In this PR we just need to merge CI validation with empty list of Inputs or any single Input. 

example of work: https://github.com/checkstyle/checkstyle/actions/runs/9909177598/job/27376661314?pr=15237

Looks like Goolge formatter enforce `{` to be on new line after `case`, we will investigate separately to follow or not this formatting. It is unrelated to FallThrough so I am ok to merge such changes. 
but update for comment location is good catch from it.
```
$ git diff
diff --git a/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4842fallthrough/InputFallThrough.java b/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4842fallthrough/InputFallThrough.java
index eda4f11..88f3130 100644
--- a/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4842fallthrough/InputFallThrough.java
+++ b/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4842fallthrough/InputFallThrough.java
@@ -20,21 +20,26 @@ public class InputFallThrough {
           throw new RuntimeException("");
         case 6:
           continue;
-        case 7: {
-          break;
-        }
-        case 8: {
-          return;
-        }
-        case 9: {
-          throw new RuntimeException("");
-        }
-        case 10: {
-          continue;
-        }
-        case 11: {
-          i++;
-        }
+        case 7:
+          {
+            break;
+          }
+        case 8:
+          {
+            return;
+          }
+        case 9:
+          {
+            throw new RuntimeException("");
+          }
+        case 10:
+          {
+            continue;
+          }
+        case 11:
+          {
+            i++;
+          }
         case 12: // violation 'Fall through from previous branch of the switch statement.'
           if (false) {
             break;
@@ -152,22 +157,27 @@ public class InputFallThrough {
           throw new RuntimeException("");
         case 6:
           continue;
-        case 7: {
-          break;
-        }
-        case 8: {
-          return;
-        }
-        case 9: {
-          throw new RuntimeException("");
-        }
-        case 10: {
-          continue;
-        }
-        case 11: {
-          i++;
-        }
-        // fallthru
+        case 7:
+          {
+            break;
+          }
+        case 8:
+          {
+            return;
+          }
+        case 9:
+          {
+            throw new RuntimeException("");
+          }
+        case 10:
+          {
+            continue;
+          }
+        case 11:
+          {
+            i++;
+          }
+          // fallthru
         case 12:
           if (false) {
             break;
@@ -253,7 +263,7 @@ public class InputFallThrough {
           }
         case 24:
           i++;
-        /* fallthru */ case 25:
+          /* fallthru */ case 25:
           i++;
           break;
 
@@ -285,13 +295,14 @@ public class InputFallThrough {
         case 1:
           i++;
           // fallthru
-        case 2: {
-          i++;
-        }
-        // fallthru
+        case 2:
+          {
+            i++;
+          }
+          // fallthru
         case 3:
           i++;
-        /* fallthru */ case 4:
+          /* fallthru */ case 4:
           break;
         case 5:
           i++;
@@ -312,7 +323,7 @@ public class InputFallThrough {
           /* fallthru */
         case 2:
           i++;
-        /* fallthru */ case 3:
+          /* fallthru */ case 3:
           break;
         case 4:
           i++;
@@ -333,7 +344,7 @@ public class InputFallThrough {
           /*fallthru*/
         case 2:
           i++;
-        /*fallthru*/ case 3:
+          /*fallthru*/ case 3:
           break;
         case 4:
           i++;
@@ -354,7 +365,7 @@ public class InputFallThrough {
           /* falls through */
         case 2:
           i++;
-        /* falls through */ case 3:
+          /* falls through */ case 3:
           break;
         case 4:
           i++;
@@ -375,7 +386,7 @@ public class InputFallThrough {
           /* Continue with next case.  */
         case 2: // violation 'Fall through from previous branch of the switch statement.'
           i++;
-        /* Continue with next case. */ case 3:
+          /* Continue with next case. */ case 3:
           // violation above 'Fall through from previous branch of the switch statement.'
           break;
         case 4:
```

command to get all:
`find src/it/resources/com/google/checkstyle/test/ -name "Input*.java" | sed "s|src/it/resources/com/google/checkstyle/test/||"`

### CODE REVIEW

**Comment by romani** [comment]
@Zopsss , please investigate if formatter enforcement is valid.

formatter is not opinionated, we just need to know if `{` on same line with `case` is not allowed by stye guide.

**Comment by Zopsss** [comment]
> Looks like Goolge formatter enforce `{` to be on new line after `case`

This seems correct formatting by google formatter.

> we just need to know if `{` on same line with `case` is not allowed by stye guide.

`{` should be present on the new line as formatted by the formatter, this formatting is done based on the section [4.1.2 Nonempty blocks: K & R style](https://google.github.io/styleguide/javaguide.html#s4.1.2-blocks-k-r-style):

> Exception: In places where these rules allow a single statement ending with a semicolon (;), a block of statements can appear, and the opening brace of this block is preceded by a line break. Blocks like these are typically introduced to limit the scope of local variables, for example inside switch statements.

This exception parts says that the `{` of block statements ( in our case the block statement followed by `case` ) should be present on the new line. 

This "exception" part was newly added in the latest version, as of the current version we're not following this newly added "exception" part. 

---

Example:

**TestOpeningCurlyBracePlacement.java**
```java
/** Some javadoc. */
public class TestOpeningCurlyBracePlacement {
  /** Some javadoc. */
  public static void main(String[] args) {
    int xyz = 10;

    switch (xyz) {
      case 1: {
        xyz++;
        break;
      }
      case 2: {
        xyz++;
        break;
      }
      case 3: {
        xyz++;
        break;
      }
      default:
        break;
    }
  }
}
```

**Output:**

```
java -jar .\checkstyle-10.17.0-all.jar -c .\google_checks.xml .\TestOpeningCurlyBracePlacement.java
Starting audit...
Audit done.
```

Code formatted by google-java-format:

```java
/** Some javadoc. */
public class TestOpeningCurlyBracePlacement {
  /** Some javadoc. */
  public static void main(String[] args) {
    int xyz = 10;

    switch (xyz) {
      case 1:
        {
          xyz++;
          break;
        }
      case 2:
        {
          xyz++;
          break;
        }
      case 3:
        {
          xyz++;
          break;
        }
      default:
        break;
    }
  }
}
```

**Output:**

```
java -jar .\checkstyle-10.17.0-all.jar -c .\google_checks.xml .\TestOpeningCurlyBracePlacement.java
Starting audit...
[WARN] C:\checkstyle testing\TestingGoogleStyle\.\TestOpeningCurlyBracePlacement.java:9:9: '{' at column 9 should be on the previous line. [LeftCurly]
[WARN] C:\checkstyle testing\TestingGoogleStyle\.\TestOpeningCurlyBracePlacement.java:14:9: '{' at column 9 should be on the previous line. [LeftCurly]
[WARN] C:\checkstyle testing\TestingGoogleStyle\.\TestOpeningCurlyBracePlacement.java:19:9: '{' at column 9 should be on the previous line. [LeftCurly]
Audit done.
```

All these warnings are false positives, the formatted code is correct based on the exception part.

---

Initially, I had doubt about this newly added exception part, so I made a comment about it on the diff PR ( [comment](https://github.com/checkstyle/checkstyle/pull/14901/files#r1621156853) ), @romani also asked question about this on google java style guide github: https://github.com/google/styleguide/commit/49488412b2f59843fce2433bd834a3fd9700604e#r143131006. 

This comment by the creator of google-java-format also talks about this false positive we have: https://github.com/checkstyle/checkstyle/issues/14294#issuecomment-1904385363

**Comment by romani** [comment]
Oh, yes, this is our most favorite point in change of style guide https://github.com/checkstyle/checkstyle/pull/14901#discussion_r1621156853

@rnveach , you were interested in this nuance, this is exactly it. Style guide updated to match what is implemented in Google java formatter.

**Review by nrmancuso — CHANGES_REQUESTED** [review]
We are heavily use Github actions to the point that I have seen workflows queued within a single job, can we put this in some other CI, and use a script to download the jar? This (using a script for everything) would be better anyway, to facilitate local execution.

**Comment by romani** [comment]
We can, but I select GitHub action to avoid authorization issues to download jar from GitHub releases, it is not maven artifact.

**Comment by nrmancuso** [comment]
```
➜  tmp curl -L -O https://github.com/google/google-java-format/releases/download/v1.22.0/google-java-format-1.22.0-all-deps.jar 
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 3527k  100 3527k    0     0  2372k      0  0:00:01  0:00:01 --:--:-- 3771k
➜  tmp ls 
google-java-format-1.22.0-all-deps.jar

```

Ideally we would just make a simple docker image for this to make it very easy for anyone to run and avoid downloading altogether.

See https://github.com/vandmo/google-java-format/blob/master/Dockerfile

**Comment by romani** [comment]
all our release scripts were forced to have github tokens as after some time or ...... it stop to work. I do not know how Github made decistion on this, but this is why all scripts use tokens - to avoid instabilities.

**Comment by nrmancuso** [comment]
After reviewing this tool, it is not clear to me how we can suppress certain formatting or customize anything. Is this not possible?

**Review by nrmancuso — APPROVED** [review]
I am good with what we have here, I posted a question above though.

**Comment by rnveach** [comment]
https://github.com/checkstyle/checkstyle/issues/15218#issuecomment-2226775850 says to come back to this in 3 days.

**Comment by romani** [comment]
> we can suppress certain formatting or customize anything. Is this not possible?

no customization, single way of formatting.
if we use it as library we can define formatting for some lines , not whole file. But I am not sure how it will help us now, we need to practice it first for some time to get experience

**Comment by romani** [comment]
@rnveach , as soon as you approve PR, I will make it green and merge able.
formatting of Inputs already in progress it will help us a lot to merge this sooner.

**Comment by romani** [comment]
I merged https://github.com/checkstyle/checkstyle/pull/15248 , so we have example of fully formatted file now.

File: .github/workflows/google-java-format.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+          java-version: 11
+          distribution: 'temurin'
+
+      - name: Checkout Pull Request Code
+        uses: actions/checkout@v4
+
+      - uses: robinraju/release-downloader@v1
+        with:
+          repository: 'google/google-java-format'
+          tag: 'v1.22.0'
```
Comment (rdiachenko) [inline .github/workflows/google-java-format.yml:39]: `1.22.0` is used 3 times in this file, does it make sense to move it into env var?

Comment (romani) [reply .github/workflows/google-java-format.yml]: done

**Review by rdiachenko — APPROVED** [review]
lgtm

**Review by rnveach — CHANGES_REQUESTED** [review]
> I will make it ... merge able.

Please do this.

> All these warnings are false positives, the formatted code is correct based on the exception part.

If we have identified this as an issue, please point to issue to resolve this in our google style.

> case 24:
i++;
`- /* fallthru */ case 25:`
`+ /* fallthru */ case 25:`

This doesn't look like a valid change to me.

**Comment by romani** [comment]
> please point to issue to resolve this in our google config.

There is already link at https://github.com/checkstyle/checkstyle/pull/15237#issuecomment-2226198183
but here it is one mote time
https://github.com/checkstyle/checkstyle/issues/14294
to make it more clear. 

>This doesn't look like a valid change to me.

right now code looks like 
https://github.com/checkstyle/checkstyle/blob/a93024d6ba68a2f8bb65de689ab39c99d5e96045/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4842fallthrough/InputFallThrough.java#L254-L258

**Comment by romani** [comment]
@Zopsss , please investgate failures at CI and suggest fix.

**Comment by rnveach** [comment]
> All these warnings are false positives
please point to issue to resolve this in our google config.
There is already link at https://github.com/checkstyle/checkstyle/pull/15237#issuecomment-2226198183
but here it is one mote time
https://github.com/checkstyle/checkstyle/issues/14294
to make it more clear.

This is not an indentation issue. This is a new line issue.

I am expecting to see an issue on making left curlies in a case block always start on a new line as google defines. It was stated above this is an issue. This GSoC project is about updating our config to the latest google style and this is listed in the google style, so this seems part of this project.

**Comment by romani** [comment]
@Zopsss , please pull this PR and find why  violations are not matching.

```
[ERROR]   FallThroughCommentedTest.testFallThrough:35->
AbstractGoogleModuleTestSupport.verifyWithWholeConfig:149->
AbstractItModuleTestSupport.verifyWithItConfig:471->
AbstractItModuleTestSupport.verifyViolations:549 Actual and expected violations differ.

expected to match: 394:(?:\d+:)?\s.*incorrect indentation level.*
but was          : 
394:42: Fall through from previous branch of the switch statement.
[INFO]
```

**Comment by rnveach** [comment]
> expected to match: 394:(?:\d+:)?\s.*incorrect indentation level.*
but was          : 
394:42: Fall through from previous branch of the switch statement.

Violation message is different. You changed indentation so the violation probably went away.

**Comment by Zopsss** [comment]
> @Zopsss , please pull this PR and find why violations are not matching.
 >
> expected to match: 394:(?:\d+:)?\s.*incorrect indentation level.*
> but was          : 
> 394:42: Fall through from previous branch of the switch statement.

You corrected the indentation level so there's no violation for it

**Comment by romani** [comment]
There was two violations I confirmed it by CLI.
It was order of violations issue, after I swapped them in input it is matching to detected.

**Comment by romani** [comment]
@rnveach , this would help us alot.

**Comment by rnveach** [comment]
I have not seen a link to a new issue at https://github.com/checkstyle/checkstyle/pull/15237#issuecomment-2231181770 .

> this would help us alot.

I don't see how. All input files have already been formatted, and from the initial responses I got at https://github.com/checkstyle/checkstyle/pull/15289 (which is also still pending) , I feel no one has really been looking at this formatter and what it does. I am not really inclined to approve this PR.

**Comment by romani** [comment]
Here is discussion and uncertainty that we can avoid if we use formatter enforced on some Inputs.

**Comment by romani** [comment]
>I have not seen a link to a new issue at

https://github.com/checkstyle/checkstyle/issues/15324

>feel no one has really been looking at this formatter and what it does.

We do in each PR, there are comments, we also found that this tool is actually do refactoring, not only formatting.

**Comment by romani** [comment]
one more example: https://github.com/checkstyle/checkstyle/pull/15372 , this Input is not about formatting at all, so better to follow  general google style guide formatting rules that formatter is offering. If in future there will diff from formatter  that we do not agree with we can: remove it from autoformatting for this file OR follow auto-formatting and put violation line (with comment `until ####` to issue in our tracker or google-java-format traker)

**Comment by romani** [comment]
>This doesn't look like a valid change to me.

Issue is reported https://github.com/google/google-java-format/issues/1123
as it is fixed, released, and we update version in this action, it will auto format back to nomal.

But still for module FallThrough this formatting does not matter.

**Comment by romani** [comment]
I pushed a bit more auto updates, to show that we lost some cases in auto formating.
Indentation module is not in strict mode for extra spacing  (from 4 minimum on wrap) is ok by our config.
BUT formatter can not be flexible, and doing it in single way.
in "rule64finalizers/InputNoFinalizer" we are ok with any identation, so auto format is preferable.

**Comment by romani** [comment]
I did one more push with initial exclude list.
in next PRs I will make list a bit more precise. 

`./.ci/google-java-format.sh .ci-temp/google-java-format-1.22.0-all-deps.jar`
`mvn failsafe:integration-test`

**Comment by romani** [comment]
@rnveach, please review this PR, 
We need this to finalize migration phase of project.

**Comment by romani** [comment]
another example of auto format - benefit:
4.8.8 Numeric Literals 	[UpperEll](https://checkstyle.org/checks/misc/upperell.html#UpperEll)

confusion around:
https://github.com/checkstyle/checkstyle/issues/15357#issuecomment-2260576956

But we will extend auto formatting to Inputs only where it does not matter.
For this Input, formatter does not update `l` to `L`, so we can use formatting and violation from checkstyle will stay.

**Comment by romani** [comment]
@rnveach , please review.

**Comment by romani** [comment]
Ohhh, I tried to rebase ...,

**Pull request merged**